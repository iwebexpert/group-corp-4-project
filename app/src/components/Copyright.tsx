import React from 'react'
import Typography, { TypographyProps } from '@mui/material/Typography';
import Link from '@mui/material/Link';

export default function Copyright(props: TypographyProps) {
  return (
    <Typography variant="body2" color="text.secondary" align="center" {...props}>
      {'Copyright © '}
      <Link color="inherit" href="https://mui.com/">
        Учебный проект
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}
