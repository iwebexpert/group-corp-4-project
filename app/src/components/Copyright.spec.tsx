import React from 'react'
import Link from '@mui/material/Link';
import Copyright from './Copyright'

import * as renderer from 'react-test-renderer'
import {test, expect} from '@jest/globals'

import {shallow, mount} from 'enzyme'

test('Copyright should be rendered', () => {
  // const a = 3
  // expect(a).toBe(3)

  const testRenderer = renderer.create(<Copyright sx={{mt: 1}} />)
  const treeJson = testRenderer.toJSON()
  expect(treeJson).toMatchSnapshot()

  const testInstance = testRenderer.root
  expect(testInstance.findByType(Link).props.href).toBe('https://mui.com/')
})

test('Copyright should be rendered as ForwardRef(Link)', () => {
  const tree = shallow(<Copyright sx={{mt: 1}} />)
  const aTag = tree.find('ForwardRef(Link)')

  // console.log(tree.debug())

  expect(aTag.length).toBe(1)
})

test('Copyright should be rendered as link', () => {
  const tree = mount(<Copyright sx={{mt: 1}} />)
  const aTag = tree.find('a')

  // console.log(tree.debug())

  expect(aTag.length).toBe(1)
})