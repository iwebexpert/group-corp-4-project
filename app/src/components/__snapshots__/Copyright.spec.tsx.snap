// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright should be rendered 1`] = `
<p
  className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-pi4kka-MuiTypography-root"
>
  Copyright © 
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
    href="https://mui.com/"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Учебный проект
  </a>
   
  2022
  .
</p>
`;
